src/pyfmi/fmi_util.pyx

```diff
@@ -1000,99 +1000,81 @@ cdef class DumpData:
-            r = self.model.get_real(self.real_var_ref)
-            i = self.model.get_integer(self.int_var_ref)
-            b = self.model.get_boolean(self.bool_var_ref)
-            
-            for k in range(self.rs):
-                self.data[k] = r[k]
-            for k in range(self.rs, self.ris):
-                self.data[k] = i[k-self.rs]
-            for k in range(self.ris, self.ribs):
-                self.data[k] = b[k-self.ris]
-            
-            return self._create_string(self.data)
+    return data
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def read_name_list(file_name, int file_position, int sizeof_type, int nbr_variables, int max_length):
+    cdef int i = 0, j = 0, py3, need_replace = 0
+    cdef FILE* cfile
+    cdef char *tmp = <char*>FMIL.calloc(max_length,sizeof(char))
+    cdef bytes py_str
+    cdef dict data = {}
     
-    cpdef _create_string(self, np.ndarray[double, ndim=1, mode='c'] data):
-        cdef int j, used_length_cache, cn = 0
-        cdef char temp_str[4094]
-        cdef char* temp_str_ptr = &temp_str[0]
-        
-        used_length_cache = 0
-        for j in range(0, self.nvariables-5, 5):
-        #    self.cache[j] = " %.14E"%self.data[self.data_order[j]]
-            #temp_str_ptr += FMIL.sprintf(temp_str_ptr, " %.14E", <double>data[self.data_order[j]])
-            #cn += FMIL.snprintf(temp_str_ptr+cn, 4094, " %.14E", <double>data[self.data_order[j]])
-            cn += FMIL.snprintf(temp_str_ptr+cn, 4094, " %.14E %.14E %.14E %.14E %.14E", 
-                                                <double>data[self.data_order[j]],
-                                                <double>data[self.data_order[j+1]],
-                                                <double>data[self.data_order[j+2]],
-                                                <double>data[self.data_order[j+3]],
-                                                <double>data[self.data_order[j+4]])
-            if j > 0 and not (j) % 150:
-                self.cache[used_length_cache] = temp_str
-                used_length_cache += 1
-                cn = 0
-                temp_str_ptr = &temp_str[0]
-        else:
-            for j in range(j, self.nvariables-1):
-                cn += FMIL.snprintf(temp_str_ptr+cn, 4094, " %.14E", <double>data[self.data_order[j]])
-        if j % 150:
-            self.cache[used_length_cache] = temp_str[:cn]
-            used_length_cache += 1
-            
-        return self._join_string(used_length_cache)
+    if python3_flag:
+        py3 = 1
+    else:
+        py3 = 0
+
+    cfile = fopen(file_name, 'rb')
     
-    cpdef _join_string(self, int used_length_cache):
-        cdef int j
+    fseek(cfile, file_position, 0)
+    for i in range(nbr_variables):
+        fread(<void*>(tmp), max_length, 1, cfile)
+        py_str = tmp
         
-        return "%.14E "%self.model.time + ' '.join(self.cache[:used_length_cache])
-
+        if i == 0:
```

Comment:
Why should this only hold true for the first variable read?

Reply:
It can hold true for others but if it does it means that the length is less than four. And if so then there is not problem as it won't cause any large overhead.

Reply:
I think it would be worth to have a comment on this in the code.